Feature/script update

jkuang-exploro opened the pull request:
Modularized forecast script reproducing the develop branch.

All vital components of the modularized forecast scripts are in place for a 
successful ATM-only experiment replicating the develop branch. Please 
comment on how could this PR be tested, as well as the test plan of the 
"final" merge toward the develop.

Issue tracker: #365 and #366 

Comment: I think maybe a specific hashtag should be used instead of the 
vague "develop branch"

Jian Kuang's test contents for the modularized script:
ROTDIR: /work/noaa/marine/Jian.Kuang/comroot/modular-192
EXPDIR: /work/noaa/marine/Jian.Kuang/expdir/modular-192
RUNDIR: /work/noaa/stmp/jkuang/RUNDIRS/modular-192/2020090118/gfs/fcst.143393

07/31 test contents, this run is NOT completed successfully, due to the following error which also appear in feature/coupled-sprint branch as of that date:

FATAL from PE     0: mpp_domains_define.inc: At least one pe in pelist is not used by any tile in the mosaic

Repo: /work/noaa/marine/Jian.Kuang/coupled-sprint
EXPDIR: /work/noaa/marine/Jian.Kuang/expdir/modular_0731
ROTDIR: /work/noaa/marine/Jian.Kuang/comroot/modular_0731
RUNDIR: /work/noaa/stmp/jkuang/RUNDIRS/modular_0731/2020090118/gfs/fcst.355316

Commit:
files intake from existing coupled-crow branch

Commit:
incremental update of modularized script

Commit:
incremental update

Commit:
incremental update

Commit:
incremental update

jkuang-exploro commented:
@aerorahul This is the 1st version that I put together. All vital components are in place for a successful ATM-only experiment replicating the develop branch. I made it a draft because it wasn't tested yet.

WalterKolczynski-NOAA commented:
Please start following global-workflow commit message guidelines (see https://github.com/NOAA-EMC/global-workflow/wiki/Development-of-Global-Workflow#commit-message-standards)

jkuang-exploro commented:
@WalterKolczynski-NOAA Appreciate bringing this up. I updated the description.

JessicaMeixner-NOAA commented:
@JianKuang-UMD this PR is already out of date with feature/coupled-crow after a PR from @WalterKolczynski-NOAA was merged yesterday.  There is one (at most 2) more PRs from @WalterKolczynski-NOAA which has changes needed for p7 but actually will likely help by adding features in the forecast script that are in the develop branch but not in feature/coupled-crow yet.

jkuang-exploro commented:
@JessicaMeixner-NOAA So should I update this PR now or after the other upcoming PR goes into coupled-crow?

jkuang-exploro commented:
@WalterKolczynski-NOAA Should I update this PR now or after the other upcoming PR goes into coupled-crow?

WalterKolczynski-NOAA commented:
I'd wait until after coupled-crow settles. The next one has the bigger forecast script changes.

aerorahul commented:
@WalterKolczynski-NOAA 
We agreed no more big commits to the coupled crow branch.

WalterKolczynski-NOAA commented:
This is still more work for the next prototype.

aerorahul commented:
How are they ready to run prototypes in 2 weeks if more work is required?
That does not make sense or there is a miscommunication.
We will discuss tomorrow at the review.

WalterKolczynski-NOAA commented:
We've said the whole time we're trying to have things ready by the end of this week, though now it may slip to Monday.

Commit:
namelist update, set up script match develop

KateFriedman-NOAA commented:
I ran the develop branch on Orion and compared my output to Jian's run with the modular forecast script. Same ICs used for both runs. There is a reproducibility issue with this hash of ufs-weather-model but the forecast output up through f15 matched. See comparison below.

```
Orion-login-4[27] /work/noaa/stmp/kfriedma/comrot/fcsdev/gfs.20200901/18/atmos$ compare_ncfile.py /work/noaa/marine/Jian.Kuang/comroot/modular-192/gfs.20200901/18/atmos/gfs.t18z.atmf015.nc gfs.t18z.atmf015.nc
grid_xt min/max 1=0.0,359.53125 min/max 2=0.0,359.53125 max abs diff=0.0000000000
lon min/max 1=0.0,359.53125 min/max 2=0.0,359.53125 max abs diff=0.0000000000
grid_yt min/max 1=-89.64164807259336,89.64164807259336 min/max 2=-89.64164807259336,89.64164807259336 max abs diff=0.0000000000
lat min/max 1=-89.64164807259336,89.64164807259336 min/max 2=-89.64164807259336,89.64164807259336 max abs diff=0.0000000000
pfull min/max 1=0.012781459,998.7807 min/max 2=0.012781459,998.7807 max abs diff=0.0000000000
phalf min/max 1=0.00999,1000.0 min/max 2=0.00999,1000.0 max abs diff=0.0000000000
time min/max 1=15.0,15.0 min/max 2=15.0,15.0 max abs diff=0.0000000000
cld_amt min/max 1=0.0,1.0000001 min/max 2=0.0,1.0000001 max abs diff=0.0000000000
clwmr min/max 1=-6.5368904e-20,0.001467026 min/max 2=-6.5368904e-20,0.001467026 max abs diff=0.0000000000
delz min/max 1=-3640.2764,-15.929722 min/max 2=-3640.2764,-15.929722 max abs diff=0.0000000000
dpres min/max 1=0.6059999,1763.2124 min/max 2=0.6059999,1763.2124 max abs diff=0.0000000000
dzdt min/max 1=-2.1237776,1.8675699 min/max 2=-2.1237776,1.8675699 max abs diff=0.0000000000
grle min/max 1=-1.850584e-20,0.0018258409 min/max 2=-1.850584e-20,0.0018258409 max abs diff=0.0000000000
hgtsfc min/max 1=-55.19578,5677.363 min/max 2=-55.19578,5677.363 max abs diff=0.0000000000
icmr min/max 1=-8.362246e-21,0.00076990423 min/max 2=-8.362246e-21,0.00076990423 max abs diff=0.0000000000
o3mr min/max 1=1.9768066e-08,1.7678107e-05 min/max 2=1.9768066e-08,1.7678107e-05 max abs diff=0.0000000000
pressfc min/max 1=51402.188,103400.016 min/max 2=51402.188,103400.016 max abs diff=0.0000000000
rwmr min/max 1=-6.4957633e-21,0.0026575255 min/max 2=-6.4957633e-21,0.0026575255 max abs diff=0.0000000000
snmr min/max 1=-1.2146279e-19,0.004904586 min/max 2=-1.2146279e-19,0.004904586 max abs diff=0.0000000000
spfh min/max 1=6.052613e-08,0.022263074 min/max 2=6.052613e-08,0.022263074 max abs diff=0.0000000000
tmp min/max 1=180.51543,317.53638 min/max 2=180.51543,317.53638 max abs diff=0.0000000000
ugrd min/max 1=-72.20959,159.68596 min/max 2=-72.20959,159.68596 max abs diff=0.0000000000
vgrd min/max 1=-84.675446,117.3277 min/max 2=-84.675446,117.3277 max abs diff=0.0000000000
```
```
Orion-login-4[28] /work/noaa/stmp/kfriedma/comrot/fcsdev/gfs.20200901/18/atmos$ compare_ncfile.py /work/noaa/marine/Jian.Kuang/comroot/modular-192/gfs.20200901/18/atmos/gfs.t18z.sfcf015.nc gfs.t18z.sfcf015.nc
grid_xt min/max 1=0.0,359.53125 min/max 2=0.0,359.53125 max abs diff=0.0000000000
lon min/max 1=0.0,359.53125 min/max 2=0.0,359.53125 max abs diff=0.0000000000
grid_yt min/max 1=-89.64164807259336,89.64164807259336 min/max 2=-89.64164807259336,89.64164807259336 max abs diff=0.0000000000
lat min/max 1=-89.64164807259336,89.64164807259336 min/max 2=-89.64164807259336,89.64164807259336 max abs diff=0.0000000000
time min/max 1=15.0,15.0 min/max 2=15.0,15.0 max abs diff=0.0000000000
acond min/max 1=7.6655255e-05,0.37537307 min/max 2=7.6655255e-05,0.37537307 max abs diff=0.0000000000
albdo_ave min/max 1=0.0,87.2877 min/max 2=0.0,87.2877 max abs diff=0.0000000000
alnsf min/max 1=0.01905039,0.6540536 min/max 2=0.01905039,0.6540536 max abs diff=0.0000000000
alnwf min/max 1=0.018865336,0.65163565 min/max 2=0.018865336,0.65163565 max abs diff=0.0000000000
alvsf min/max 1=0.014070779,0.6519257 min/max 2=0.014070779,0.6519257 max abs diff=0.0000000000
alvwf min/max 1=0.013472418,0.6519257 min/max 2=0.013472418,0.6519257 max abs diff=0.0000000000
aod550 min/max 1=0.009115649,1.481971 min/max 2=0.009115649,1.481971 max abs diff=0.0000000000
bc_aod550 min/max 1=0.0008537131,0.104453824 min/max 2=0.0008537131,0.104453824 max abs diff=0.0000000000
c0 min/max 1=0.0,0.077707395 min/max 2=0.0,0.077707395 max abs diff=0.0000000000
cd min/max 1=-580.73987,0.0 min/max 2=-580.73987,0.0 max abs diff=0.0000000000
cduvb_ave min/max 1=0.0,22.131285 min/max 2=0.0,22.131285 max abs diff=0.0000000000
cnvprcp min/max 1=0.0,0.0010432269 min/max 2=0.0,0.0010432269 max abs diff=0.0000000000
cnwat min/max 1=0.0,0.5 min/max 2=0.0,0.5 max abs diff=0.0000000000
cpofp min/max 1=0.0,1.0000001 min/max 2=0.0,1.0000001 max abs diff=0.0000000000
cprat_ave min/max 1=0.0,0.0007404335 min/max 2=0.0,0.0007404335 max abs diff=0.0000000000
cpratb_ave min/max 1=0.0,0.0013801329 min/max 2=0.0,0.0013801329 max abs diff=0.0000000000
crain min/max 1=0.0,1.0 min/max 2=0.0,1.0 max abs diff=0.0000000000
csdlf min/max 1=69.70384,474.11243 min/max 2=69.70384,474.11243 max abs diff=0.0000000000
csdsf min/max 1=0.0,1010.89355 min/max 2=0.0,1010.89355 max abs diff=0.0000000000
csulf min/max 1=115.540825,676.82556 min/max 2=115.540825,676.82556 max abs diff=0.0000000000
csulftoa min/max 1=118.66877,367.13226 min/max 2=118.66877,367.13226 max abs diff=0.0000000000
csusf min/max 1=0.0,770.80884 min/max 2=0.0,770.80884 max abs diff=0.0000000000
csusftoa min/max 1=0.0,713.62964 min/max 2=0.0,713.62964 max abs diff=0.0000000000
cwork_aveclm min/max 1=0.0,1561.4344 min/max 2=0.0,1561.4344 max abs diff=0.0000000000
dconv min/max 1=0.0,6.244938 min/max 2=0.0,6.244938 max abs diff=0.0000000000
dlwrf min/max 1=70.71169,501.53412 min/max 2=70.71169,501.53412 max abs diff=0.0000000000
dlwrf_ave min/max 1=69.91925,491.6324 min/max 2=69.91925,491.6324 max abs diff=0.0000000000
dswrf min/max 1=0.0,1060.8348 min/max 2=0.0,1060.8348 max abs diff=0.0000000000
dswrf_ave min/max 1=0.0,1010.86163 min/max 2=0.0,1010.86163 max abs diff=0.0000000000
dswrf_avetoa min/max 1=0.0,1303.6658 min/max 2=0.0,1303.6658 max abs diff=0.0000000000
dtcool min/max 1=0.0,1.4649872 min/max 2=0.0,1.4649872 max abs diff=0.0000000000
du_aod550 min/max 1=0.00072498986,0.547553 min/max 2=0.00072498986,0.547553 max abs diff=0.0000000000
duvb_ave min/max 1=0.0,21.922491 min/max 2=0.0,21.922491 max abs diff=0.0000000000
evbs_ave min/max 1=0.0,400.80872 min/max 2=0.0,400.80872 max abs diff=0.0000000000
evcw_ave min/max 1=0.0,546.1592 min/max 2=0.0,546.1592 max abs diff=0.0000000000
f10m min/max 1=0.9316632,1.312024 min/max 2=0.9316632,1.312024 max abs diff=0.0000000000
facsf min/max 1=0.0,1.0 min/max 2=0.0,1.0 max abs diff=0.0000000000
facwf min/max 1=-4.2632563e-16,1.0 min/max 2=-4.2632563e-16,1.0 max abs diff=0.0000000000
ffhh min/max 1=1.181344,52.21402 min/max 2=1.181344,52.21402 max abs diff=0.0000000000
ffmm min/max 1=1.3604072,54.366844 min/max 2=1.3604072,54.366844 max abs diff=0.0000000000
fldcp min/max 1=0.0,0.419691 min/max 2=0.0,0.419691 max abs diff=0.0000000000
fricv min/max 1=0.0073574255,2.191457 min/max 2=0.0073574255,2.191457 max abs diff=0.0000000000
gflux min/max 1=-496.3348,159.95694 min/max 2=-496.3348,159.95694 max abs diff=0.0000000000
gflux_ave min/max 1=-445.56363,161.64084 min/max 2=-445.56363,161.64084 max abs diff=0.0000000000
hgt_hyblev1 min/max 1=7.9631543,11.536963 min/max 2=7.9631543,11.536963 max abs diff=0.0000000000
hpbl min/max 1=8.056323,5687.9355 min/max 2=8.056323,5687.9355 max abs diff=0.0000000000
icec min/max 1=0.0,1.0 min/max 2=0.0,1.0 max abs diff=0.0000000000
icetk min/max 1=0.0,5.0065074 min/max 2=0.0,5.0065074 max abs diff=0.0000000000
land min/max 1=0.0,2.0 min/max 2=0.0,2.0 max abs diff=0.0000000000
lhtfl min/max 1=-65.13178,911.34393 min/max 2=-65.13178,911.34393 max abs diff=0.0000000000
lhtfl_ave min/max 1=-64.09785,840.77875 min/max 2=-64.09785,840.77875 max abs diff=0.0000000000
nbdsf_ave min/max 1=0.0,510.1645 min/max 2=0.0,510.1645 max abs diff=0.0000000000
nddsf_ave min/max 1=0.0,369.62607 min/max 2=0.0,369.62607 max abs diff=0.0000000000
oc_aod550 min/max 1=0.0019335842,1.1782336 min/max 2=0.0019335842,1.1782336 max abs diff=0.0000000000
orog min/max 1=-62.2226,5702.785 min/max 2=-62.2226,5702.785 max abs diff=0.0000000000
pevpr min/max 1=-30.324945,2574.2175 min/max 2=-30.324945,2574.2175 max abs diff=0.0000000000
pevpr_ave min/max 1=-26.324642,2615.0718 min/max 2=-26.324642,2615.0718 max abs diff=0.0000000000
prate_ave min/max 1=0.0,0.0037249955 min/max 2=0.0,0.0037249955 max abs diff=0.0000000000
prateb_ave min/max 1=0.0,0.0080602635 min/max 2=0.0,0.0080602635 max abs diff=0.0000000000
pres_avehcb min/max 1=5468.017,50491.145 min/max 2=5468.017,50491.145 max abs diff=0.0000000000
pres_avehct min/max 1=4784.438,47155.79 min/max 2=4784.438,47155.79 max abs diff=0.0000000000
pres_avelcb min/max 1=65145.93,103085.266 min/max 2=65145.93,103085.266 max abs diff=0.0000000000
pres_avelct min/max 1=64158.492,102376.65 min/max 2=64158.492,102376.65 max abs diff=0.0000000000
pres_avemcb min/max 1=40300.88,74896.36 min/max 2=40300.88,74896.36 max abs diff=0.0000000000
pres_avemct min/max 1=39157.105,73459.56 min/max 2=39157.105,73459.56 max abs diff=0.0000000000
prescnvclb min/max 1=38310.215,102175.81 min/max 2=38310.215,102175.81 max abs diff=0.0000000000
prescnvclt min/max 1=7564.2275,97245.31 min/max 2=7564.2275,97245.31 max abs diff=0.0000000000
pressfc min/max 1=51402.08,103399.96 min/max 2=51402.08,103399.96 max abs diff=0.0000000000
pwatclm min/max 1=0.1179889,92.85657 min/max 2=0.1179889,92.85657 max abs diff=0.0000000000
qrain min/max 1=-5.1567416,53.36178 min/max 2=-5.1567416,53.36178 max abs diff=0.0000000000
sbsno_ave min/max 1=0.0,146.56505 min/max 2=0.0,146.56505 max abs diff=0.0000000000
sfcr min/max 1=1e-07,2.653 min/max 2=1e-07,2.653 max abs diff=0.0000000000
sfexc min/max 1=8.3723935e-05,0.3580269 min/max 2=8.3723935e-05,0.3580269 max abs diff=0.0000000000
shdmax min/max 1=0.0,0.99 min/max 2=0.0,0.99 max abs diff=0.0000000000
shdmin min/max 1=0.0,0.8874654 min/max 2=0.0,0.8874654 max abs diff=0.0000000000
shtfl min/max 1=-303.62604,689.8421 min/max 2=-303.62604,689.8421 max abs diff=0.0000000000
shtfl_ave min/max 1=-231.70915,528.4035 min/max 2=-231.70915,528.4035 max abs diff=0.0000000000
sltyp min/max 1=0.0,9.0 min/max 2=0.0,9.0 max abs diff=0.0000000000
snoalb min/max 1=0.0,0.84000003 min/max 2=0.0,0.84000003 max abs diff=0.0000000000
snod min/max 1=0.0,2.7449734 min/max 2=0.0,2.7449734 max abs diff=0.0000000000
snohf min/max 1=-0.33085096,258.92303 min/max 2=-0.33085096,258.92303 max abs diff=0.0000000000
snowc_ave min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
soill1 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soill2 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soill3 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soill4 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soilm min/max 1=40.0,2000.0 min/max 2=40.0,2000.0 max abs diff=0.0000000000
soilt1 min/max 1=213.63599,322.24103 min/max 2=213.63599,322.24103 max abs diff=0.0000000000
soilt2 min/max 1=215.41876,316.31528 min/max 2=215.41876,316.31528 max abs diff=0.0000000000
soilt3 min/max 1=218.00589,315.588 min/max 2=218.00589,315.588 max abs diff=0.0000000000
soilt4 min/max 1=220.98828,313.90536 min/max 2=220.98828,313.90536 max abs diff=0.0000000000
soilw1 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soilw2 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soilw3 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
soilw4 min/max 1=0.02,1.0 min/max 2=0.02,1.0 max abs diff=0.0000000000
sotyp min/max 1=0.0,16.0 min/max 2=0.0,16.0 max abs diff=0.0000000000
spfh2m min/max 1=2.1114145e-05,0.022690546 min/max 2=2.1114145e-05,0.022690546 max abs diff=0.0000000000
spfh_hyblev1 min/max 1=2.2855056e-05,0.02227621 min/max 2=2.2855056e-05,0.02227621 max abs diff=0.0000000000
spfhmax_max2m min/max 1=2.2431675e-05,0.022846607 min/max 2=2.2431675e-05,0.022846607 max abs diff=0.0000000000
spfhmin_min2m min/max 1=1.8100864e-05,0.0224638 min/max 2=1.8100864e-05,0.0224638 max abs diff=0.0000000000
ss_aod550 min/max 1=0.00016142667,0.16363867 min/max 2=0.00016142667,0.16363867 max abs diff=0.0000000000
ssrun_acc min/max 1=0.0,72.80799 min/max 2=0.0,72.80799 max abs diff=0.0000000000
su_aod550 min/max 1=0.0035001924,0.626611 min/max 2=0.0035001924,0.626611 max abs diff=0.0000000000
sunsd_acc min/max 1=0.0,10800.001 min/max 2=0.0,10800.001 max abs diff=0.0000000000
tcdc_avebndcl min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
tcdc_aveclm min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
tcdc_avehcl min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
tcdc_avelcl min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
tcdc_avemcl min/max 1=0.0,100.0 min/max 2=0.0,100.0 max abs diff=0.0000000000
tcdccnvcl min/max 1=0.0,80.0 min/max 2=0.0,80.0 max abs diff=0.0000000000
tg3 min/max 1=215.48914,304.02972 min/max 2=215.48914,304.02972 max abs diff=0.0000000000
tisfc min/max 1=214.31944,345.20267 min/max 2=214.31944,345.20267 max abs diff=0.0000000000
tmax_max2m min/max 1=218.98985,318.39227 min/max 2=218.98985,318.39227 max abs diff=0.0000000000
tmin_min2m min/max 1=217.22723,313.29218 min/max 2=217.22723,313.29218 max abs diff=0.0000000000
tmp2m min/max 1=218.98985,318.39227 min/max 2=218.98985,318.39227 max abs diff=0.0000000000
tmp_avehct min/max 1=181.49324,261.49017 min/max 2=181.49324,261.49017 max abs diff=0.0000000000
tmp_avelct min/max 1=225.09671,299.7898 min/max 2=225.09671,299.7898 max abs diff=0.0000000000
tmp_avemct min/max 1=220.84262,282.62845 min/max 2=220.84262,282.62845 max abs diff=0.0000000000
tmp_hyblev1 min/max 1=219.86157,317.33945 min/max 2=219.86157,317.33945 max abs diff=0.0000000000
tmpsfc min/max 1=214.31944,345.20267 min/max 2=214.31944,345.20267 max abs diff=0.0000000000
tprcp min/max 1=0.0,0.003266652 min/max 2=0.0,0.003266652 max abs diff=0.0000000000
trans_ave min/max 1=0.0,540.5403 min/max 2=0.0,540.5403 max abs diff=0.0000000000
tref min/max 1=212.89056,336.36594 min/max 2=212.89056,336.36594 max abs diff=0.0000000000
u-gwd_ave min/max 1=-4.880547,10.564625 min/max 2=-4.880547,10.564625 max abs diff=0.0000000000
uflx_ave min/max 1=-3.3018656,3.1014175 min/max 2=-3.3018656,3.1014175 max abs diff=0.0000000000
ugrd10m min/max 1=-36.08228,38.67949 min/max 2=-36.08228,38.67949 max abs diff=0.0000000000
ugrd_hyblev1 min/max 1=-36.50257,39.120895 min/max 2=-36.50257,39.120895 max abs diff=0.0000000000
ulwrf min/max 1=120.312096,758.6273 min/max 2=120.312096,758.6273 max abs diff=0.0000000000
ulwrf_ave min/max 1=116.28082,707.26935 min/max 2=116.28082,707.26935 max abs diff=0.0000000000
ulwrf_avetoa min/max 1=84.10931,367.13226 min/max 2=84.10931,367.13226 max abs diff=0.0000000000
uswrf min/max 1=0.0,645.65546 min/max 2=0.0,645.65546 max abs diff=0.0000000000
uswrf_ave min/max 1=0.0,759.22595 min/max 2=0.0,759.22595 max abs diff=0.0000000000
uswrf_avetoa min/max 1=0.0,909.6714 min/max 2=0.0,909.6714 max abs diff=0.0000000000
v-gwd_ave min/max 1=-6.181528,8.3638525 min/max 2=-6.181528,8.3638525 max abs diff=0.0000000000
vbdsf_ave min/max 1=0.0,430.7707 min/max 2=0.0,430.7707 max abs diff=0.0000000000
vddsf_ave min/max 1=0.0,405.05283 min/max 2=0.0,405.05283 max abs diff=0.0000000000
veg min/max 1=0.0,99.0 min/max 2=0.0,99.0 max abs diff=0.0000000000
vflx_ave min/max 1=-3.4080212,2.8411288 min/max 2=-3.4080212,2.8411288 max abs diff=0.0000000000
vgrd10m min/max 1=-31.631641,38.28689 min/max 2=-31.631641,38.28689 max abs diff=0.0000000000
vgrd_hyblev1 min/max 1=-31.981686,38.738304 min/max 2=-31.981686,38.738304 max abs diff=0.0000000000
vtype min/max 1=0.0,20.0 min/max 2=0.0,20.0 max abs diff=0.0000000000
w0 min/max 1=-314.09402,0.6492158 min/max 2=-314.09402,0.6492158 max abs diff=0.0000000000
watr_acc min/max 1=0.0,73.03194 min/max 2=0.0,73.03194 max abs diff=0.0000000000
wd min/max 1=-0.98544055,336.92212 min/max 2=-0.98544055,336.92212 max abs diff=0.0000000000
weasd min/max 1=0.0,415.8015 min/max 2=0.0,415.8015 max abs diff=0.0000000000
wilt min/max 1=0.0,0.39735234 min/max 2=0.0,0.39735234 max abs diff=0.0000000000
xs min/max 1=-1.6352863,0.09205757 min/max 2=-1.6352863,0.09205757 max abs diff=0.0000000000
xt min/max 1=0.0,4.074173 min/max 2=0.0,4.074173 max abs diff=0.0000000000
xtts min/max 1=-146.40254,30.948713 min/max 2=-146.40254,30.948713 max abs diff=0.0000000000
xu min/max 1=-1.6308749,1.726182 min/max 2=-1.6308749,1.726182 max abs diff=0.0000000000
xv min/max 1=-1.5081172,1.475466 min/max 2=-1.5081172,1.475466 max abs diff=0.0000000000
xz min/max 1=0.32580042,30.0 min/max 2=0.32580042,30.0 max abs diff=0.0000000000
xzts min/max 1=-182.07448,1232531.5 min/max 2=-182.07448,1232531.5 max abs diff=0.0000000000
zc min/max 1=0.0,0.005362158 min/max 2=0.0,0.005362158 max abs diff=0.0000000000
Orion-login-4[29] /work/noaa/stmp/kfriedma/comrot/fcsdev/gfs.20200901/18/atmos$
```

aerorahul commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
 # Ignore all compiled files
 #--------------------------
+sorc/nceplibs-ncio/
```
why is this relevant to this PR?

jkuang-exploro replied:
This is a directory of some compiled objects? no?

WalterKolczynski-NOAA replied:
Leave it out. PR #375 eliminates the hack that creates this directory anyway.

aerorahul commented on ush/forecast_det.sh:
```diff
@@ -0,0 +1,82 @@
+#! /bin/sh
+
+##### 
+## "forecast_def.sh"
```
```suggestion
## "forecast_det.sh"
```

aerorahul commented on ush/forecast_det.sh:
```diff
@@ -0,0 +1,82 @@
+
+	# Determine if this is a warm start or cold start
+	if [ -f $gmemdir/RESTART/${sPDY}.${scyc}0000.coupler.res ]; then
+		export warm_start=".true."
```
please fix indents to 2 spaces.

aerorahul commented on ush/forecast_det.sh:
```diff
@@ -0,0 +1,82 @@
+	# turn IAU off for cold start
+	DOIAU_coldstart=${DOIAU_coldstart:-"NO"}
+	if [ $DOIAU = "YES" -a $warm_start = ".false." ] || [ $DOIAU_coldstart = "YES" -a $warm_start = ".true." ]; then
+		export DOIAU="NO"
+		echo "turning off IAU since warm_start = $warm_start"
+		DOIAU_coldstart="YES"
+		IAU_OFFSET=0
+		sCDATE=$CDATE
+		sPDY=$PDY
+		scyc=$cyc
+		tPDY=$sPDY
+		tcyc=$cyc
```
please fix indents to 2 spaces.

aerorahul reviewed (changes requested):
The indents in the scripts are completely whacky.
Please use 2 space indents in the scripts.  4 spaces are acceptable too, but need to be consistent.
`exglobal_forecast.sh` is removed, why?  And if it is removed, how are you running it.

Please don't add new exglobal scripts with `_bk`.  What is the intention of having those?
Version control takes care of this stuff.

aerorahul commented on ush/forecast_predet.sh:
```diff
@@ -0,0 +1,299 @@
+	#ncols=$(( (${npx}-1)*(${npy}-1)*3/2 ))
+
+	# spectral truncation and regular grid resolution based on FV3 resolution
+	JCAP_CASE=$((2*res-2))
+	LONB_CASE=$((4*res))
+	LATB_CASE=$((2*res))
+	if [ $LATB_CASE -eq 192 ]; then
+		LATB_CASE=190 # berror file is at this resolution
+	fi
+
+	JCAP=${JCAP:-$JCAP_CASE}
+	LONB=${LONB:-$LONB_CASE}
+	LATB=${LATB:-$LATB_CASE}
+
+	LONB_IMO=${LONB_IMO:-$LONB_CASE}
+	LATB_JMO=${LATB_JMO:-$LATB_CASE}
```
where in the forecast model is this used?

WalterKolczynski-NOAA replied:
JCAP/LONB/LATB are still used in some of the source fix file names.

aerorahul replied:
forgot about those.  Thanks!

aerorahul commented on ush/nems.configure.atm_aer.IN:
```diff
@@ -0,0 +1,34 @@
+#############################################
```
These are templates.
Why are they in `ush`
Shouldn't they be part of `parm` or a subdirectory within?

aerorahul replied:
aah yes.  I forgot about those spectral relics.

aerorahul replied:
Ignore this comment because we will be updating this branch with f/c-c and that may cause conflicts.

aerorahul commented:
@KateFriedman-NOAA @JianKuang-UMD 
Can you post paths to your EXPDIR, ROTDIR and the RUNDIR for the forecast.

jkuang-exploro committed:
Update ush/forecast_det.sh

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

Commit:
2-space update for exglobal_forecast.sh; forecast_predet, det and
postdet

KateFriedman-NOAA commented:
For my tests:
EXPDIR: /work/noaa/global/kfriedma/expdir/fcsdev
ROTDIR: /work/noaa/stmp/kfriedma/comrot/fcsdev
Forgot to set KEEPDATA=YES so rundir was wiped at end of job....however, I killed the initial run of the gfsfcst job when I realized I set the FHMAX to be too large and that rundir from that initial attempt was not wiped. I saved the output from the initial attempt and it was matching (/work/noaa/stmp/kfriedma/comrot/fcsdev/gfs.20200901/18/atmos/SAVE_1).
RUNDIR: /work/noaa/stmp/kfriedma/RUNDIRS/fcsdev/2020090118/gfs/fcst.325028

Commit:
Merge remote-tracking branch 'origin/feature/coupled_sprint' into feature/script_update

jkuang-exploro commented:
I updated the contents of feature/coupled-sprint into this PR, which will let us keep up with the develop branch.

I created a tag named "coupled_sprint_start" pointing to the hash of coupled-crow when I initially started the merge. I will update this PR with coupled-crow branch later today.

Commit:
update script files under script/ and ush/ to the newest
version of feature/coupled-crow, tag: coupled-sprint-0728

Updated template files in parm and ush to the newest version
of feature/coupled-crow, tag: coupled-sprint-0728

Commit:
Merge remote-tracking branch 'origin/feature/coupled_sprint' into feature/script_update

JessicaMeixner-NOAA commented on scripts/exgfs_wave_prep.sh:
```diff
@@ -224,16 +224,16 @@
               ;;
```
Please revert these changes, this script is not involved w/the forecast and should remain as is for this PR.

jkuang-exploro replied:
OK. I addressed these.

JessicaMeixner-NOAA commented on scripts/exglobal_forecast.sh_bk:
```diff
@@ -0,0 +1,1422 @@
+#!/bin/ksh
```
I thought this file should be deleted?

jkuang-exploro replied:
Of course, doing it now...

Commit:
Merge branch 'feature/coupled_sprint' into feature/script_update

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,961 @@
+  #------------------------------------------------------------------
+  # make symbolic links to write forecast files directly in memdir
+  cd $DATA
+  if [ $RUN_CCPP = "YES" ]; then
```
This seems out of date with f/c-c 
It's my understanding that RUN_CCPP is now always YES so we shouldn't need this If statement here.

jkuang-exploro replied:
I know that is true for the coupled model, but what about the develop side? @KateFriedman-NOAA @WalterKolczynski-NOAA May I ask you this question: Is RUN_CCPP always YES? on the develop side?

KateFriedman-NOAA replied:
@JianKuang-UMD The develop branch sets RUN_CCPP=YES by default but I removed that variable in my commit to feature/coupled_sprint. It's my understanding that we'll be running CCPP always...can anyone confirm otherwise? We may not need RUN_CCPP or CCPP on/off toggles anymore.

WalterKolczynski-NOAA replied:
Yes, CCPP will always be on now. The build script automatically uses it now.

WalterKolczynski-NOAA replied:
Some of the changes I made to the forecast script also look at the CCPP suite definition file to determine settings.

jkuang-exploro replied:
Thanks for your comments, I removed RUN_CCPP from the scripts

KateFriedman-NOAA commented:
~~@JianKuang-UMD Please revert the removal of the parm/parm_fv3diag/diag_table_aod file, that is still used by MERRA2 in atm-only mode. Thanks!~~

WalterKolczynski-NOAA commented:
> @JianKuang-UMD Please revert the removal of the parm/parm_fv3diag/diag_table_aod file, that is still used by MERRA2 in atm-only mode. Thanks!

Don't revert. New changes that should be brought over to the forecast scripts change how this diag table works. It is now appended to the base diag_table is MERRA2 is on.

KateFriedman-NOAA commented:
> > @JianKuang-UMD Please revert the removal of the parm/parm_fv3diag/diag_table_aod file, that is still used by MERRA2 in atm-only mode. Thanks!
> 
> Don't revert. New changes that should be brought over to the forecast scripts change how this diag table works. It is now appended to the base diag_table is MERRA2 is on.

@WalterKolczynski-NOAA Ok I see, thanks for explaining! Ignore my request @JianKuang-UMD thanks!

KateFriedman-NOAA reviewed (approved):
No further comments, thanks for this work @JianKuang-UMD !

Commit:
delete _bk file

Commit:
revert changes for exgfs_wave_prep.sh

Commit:
remove "RUN_CCPP" variable

WalterKolczynski-NOAA commented on sorc/build_ufs_coupled.sh:
```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env bash
```
This file is not needed and should be removed. The unified build system already takes care of this.

jkuang-exploro replied:
Resolved

WalterKolczynski-NOAA commented on ush/cplvalidate.sh:
```diff
@@ -0,0 +1,40 @@
+return
+case $confignamevarfornems in
+	'atm') combination=.false..false..false..false..false.;;
+        'datm') combination=.true..true..false..false..false.;;
```
All of the case switches should be indented to the same level

jkuang-exploro replied:
Thanks for picking it up. Some of them are using tab key. I wasn't aware of it. Now resolved.

JessicaMeixner-NOAA commented on sorc/build_ufs_coupled.sh:
```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env bash
```
This file can be deleted as we have https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled_sprint/sorc/build_ufs.sh instead

jkuang-exploro replied:
Done. This is resolved.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,959 @@
+  FNZORC=${FNZORC:-"igbp"}
+  FNAISC=${FNAISC:-"$FIX_AM/CFSR.SEAICE.1982.2012.monthly.clim.grb"}
+  if [ $cplflx = ".false." ] ; then
+    FNALBC2=${FNALBC2:-"$FIX_AM/global_albedo4.1x1.grb"}
```
Most of these _probably_ need to be updated to use tiled fix files the same as for coupled runs. Need confirmation from someone more knowledgeable. That can be done later though.

JessicaMeixner-NOAA replied:
Agreed, I think everyone is moving to the tiled fix files not just coupled, however if this change isn't in develop having this if statement is necessary for reproducing the develop branch.

JessicaMeixner-NOAA commented on parm/parm_fv3diag/diag_table_aer:
```diff
@@ -0,0 +1,366 @@
+#20161003.00Z.C96.64bit.non-mono
```
If this file is added, why is the file: 
diag_table_cpl (https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/parm/parm_fv3diag/diag_table_cpl ) not added?  I don't think any of these are needed/required for ATM-only but I don't know why only some of the coupled and not all the coupled updates are getting added? 

Also there is a change for the coupled side in : https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/parm/parm_fv3diag/data_table but again not needed for ATM-only.

jkuang-exploro replied:
OK now added that file. I'm aware of the changes in data_table but decided to not include it because I don't know if it will affect atm-only run.

WalterKolczynski-NOAA commented on ush/nems_configure.sh:
```diff
@@ -0,0 +1,169 @@
+    else 
+      restart_interval_nems=$FHMAX 
+    fi
+        sed -i -e "s;@\[ocn_model\];$OCN_model;g" tmp1
```
Indentation of these sed lines should be uniform

jkuang-exploro replied:
Resolved

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,959 @@
+  day=$(echo $CDATE|cut -c 7-8)
+  stepsperhr=$((3600/$ICETIM))
+  nhours=$($NHOUR $CDATE ${year}010100)
+  istep0=$((nhours*stepsperhr))
```
This istep0 line is only in this branch, not f/c-c, I think this was recently removed.

jkuang-exploro replied:
Resolved.

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,959 @@
+  fi
+}
+
+GSD_in() {
```
GSD_in GSD_nml and GSD_out were all removed in f/c-c and should be removed here too.

jkuang-exploro replied:
resolved

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,959 @@
+  export wavprfx=${CDUMPwave}${WAV_MEMBER}
+  
+  #Copy initial condition files:
+  for wavGRD in $waveGRD ; do
```
The wave parts in this next section all appear to have been changed from what's in f/c-c to match what is in develop branch.  Just making a note that when we need to reconcile with reproducing f/c-c the differences here will have to be dealt with.  Nothing to change right now though.

JessicaMeixner-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -0,0 +1,299 @@
+  PARM_POST=${PARM_POST:-$HOMEgfs/parm/post}
+  
+  # Wave coupling parameter defaults to false
+  cplwav=${cplwav:-.false.}
```
This is new compared with f/c-c, cplwav defaulting to false is fine, I'm just curious why this was needed here though?  I mean there's not cplflx=false, etc. (That I saw)

jkuang-exploro replied:
This is from the develop branch. My speculation is, the config files in f/c-c always take care of the cpl switches including this one, while the config files in develop sometimes don't. So they need a default value here in the script. @KateFriedman-NOAA @WalterKolczynski-NOAA Is this line about cplwav default value actually required in the develop branch? Can we remove it?

WalterKolczynski-NOAA replied:
At the very least, this is probably the wrong location for it. It's definitely set in the config file now so it should be able to be removed.

jkuang-exploro replied:
Resolved

JessicaMeixner-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -0,0 +1,299 @@
+  HOMEgfs=${HOMEgfs:-$NWPROD}
+  FIX_DIR=${FIX_DIR:-$HOMEgfs/fix}
+  FIX_AM=${FIX_AM:-$FIX_DIR/fix_am}
+  export FIX_AER=${FIX_AER:-$FIX_DIR/fix_aer}
```
export FIX_AER=${FIX_AER:-$FIX_DIR/fix_aer}
  export FIX_LUT=${FIX_LUT:-$FIX_DIR/fix_lut}
are new from f/c-c they're also exported while none of the other variables are?  Is there a reason for this?

jkuang-exploro replied:
These are from the develop branch. @KateFriedman-NOAA @WalterKolczynski-NOAA Could you answer this question?

WalterKolczynski-NOAA replied:
We just discussed this a couple days ago. These do not need to be exported.

jkuang-exploro replied:
Resolved.

JessicaMeixner-NOAA commented on ush/get_platform.sh:
```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
```
This script does not appear to actually be used anywhere, is this actually needed?

jkuang-exploro replied:
If it's not needed in f/c-c then I will remove it from this PR.

WalterKolczynski-NOAA replied:
The load_fv3gfs_modules script uses it on the CROW side. I think Raffaele added it. I think it's a good helper script to have to stop repeating the logic in other scripts, we just need to go through and use it more extensively.

JessicaMeixner-NOAA replied:
I'm not saying it's not a good helper script, but if it's not being used, it's just extra and is not related to this PR and therefore should not be included in this PR.

Commit:
added diag_table_cpl
indentation fix in various scripts
remove build_ufs_coupled.sh

Commit:
delete get_platform
update forecast_postdet

Commit:
remove excessive export statements
remove cplwav in the script

aerorahul committed:
force to use NTASKS_TOT on Orion for forecast

Commit:
Remove CROW-related comments

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,619 @@
+  pdfcld       = ${pdfcld:-".false."}
+  fhswr        = ${FHSWR:-"3600."}
+  fhlwr        = ${FHLWR:-"3600."}
+  ialb         = ${IALB:-"1"}           ! In config.fcst
```
Are the "! In config.fcst" crow related or also valid comments w/out crow?

jkuang-exploro replied:
This is also cleaned up now.

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,619 @@
+  iopt_stc     = ${iopt_stc:-"1"}
+  debug        = ${gfs_phys_debug:-".false."}
+  nstf_name    = $nstf_name
+  nst_anl      = $nst_anl            ! In Workflow
```
Is the "! In Workflow" comment CROW related?

jkuang-exploro replied:
OK then, I will remove them altogether. These marks only exist in a small fraction of variables, AND could be out-of-dated, so won't be helpful anyway.

Commit:
further cleanup comments

WalterKolczynski-NOAA committed:
Fix PET issue when not using full node

EMSF fails for atm-only if the total number of tasks is not equal
to the number of PETs specified in the namelists. Now srun specifies
the number of tasks for the forecast job on Orion so the correct
number of tasks is used.

Refs: #365

WalterKolczynski-NOAA committed:
Merge branch 'feature/script_update' of github.com:NOAA-EMC/global-workflow into feature/script_update

Commit:
replace ufs_coupled with ufs_model in nems_configure.sh

aerorahul committed:
remove ncld from the namelist. This is a relic from IPD days

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+if [ $cpl = .true. ]; then
+  cat >> input.nml << EOF
+  dz_min =  ${dz_min:-"6"}    ! no longer in develop branch
+  psm_bc = ${psm_bc:-"0"}    ! no longer in develop branch
```
Two things here: 
1. Do we need the comment? 
2. ppm_bc defaults to 1 in f/c-c

WalterKolczynski-NOAA replied:
This is another change that was made to coupled just this week, though I can't find where I got the change from.

WalterKolczynski-NOAA replied:
Found it. Fanglin said to change it to 1 because of a change to dz_min = 2. So if the default dz_min is still 6, I'm guessing this should stay 0.

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+  ttendlim     = ${ttendlim:-0.005}
+  oz_phys      = ${oz_phys:-".false."}
+  oz_phys_2015 = ${oz_phys_2015:-".true."}
+  lsoil_lsm    = ${lsoil_lsm:-"4"}
```
in f/c-c this is just lsoil?

jkuang-exploro replied:
Currently only FV3_GFS_v16* and FV3_GSD_v0 suite have lsoil_lsm in place.

WalterKolczynski-NOAA replied:
This might need to be moved elsewhere. I remember removing it from coupled because it was no longer recognized, maybe because of Noah-MP? I'd have to go back and look.

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+  hybedmf      = ${hybedmf:-".false."}
+  satmedmf     = ${satmedmf-".true."}
+  isatmedmf    = ${isatmedmf-"1"}
+  lheatstrg    = ${lheatstrg-".false."}
```
In f/c-c: 
<   lheatstrg    = ${lheatstrg:-".true."}
<   lseaspray    = ${lseaspray:-".true."}
where as here it's: 
>   lheatstrg    = ${lheatstrg-".false."}

WalterKolczynski-NOAA replied:
This is another recent change to coupled.

WalterKolczynski-NOAA replied:
Do we know why this change was made for coupled? I'm guessing we should leave it false here.

JessicaMeixner-NOAA replied:
I'm looking at what the diffs are between f/c-c and this branch.  So to me the question is "why is this different" not why was it made in coupling.

WalterKolczynski-NOAA replied:
Well, it matters so we can determine what it should be for atm-only.

JessicaMeixner-NOAA replied:
@aerorahul had a great idea about figuring out which regression test you are trying to mimic and go with that.

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+if [ $cpl = .true. ]; then
+  cat >> input.nml << EOF
+  frac_grid    = ${FRAC_GRID:-".true."}
+  cplchm = ${cplchem:-".false."}
```
In f/c-c there is also a variable: 
fscav_aero   = ${fscav_aero:-'*:0.0'}

jkuang-exploro replied:
Currently it is there

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+  cplwav2atm   = ${cplwav2atm}       
+EOF
+fi
+
```
In f/c-c there is the if-block: 
< if [ ${DO_CA:-"YES"} = "YES" ]; then
<   cat >> input.nml << EOF
<   do_ca      = .True.
<   ca_sgs     = ${ca_sgs:-".True."}
<   nca        = ${nca:-"1"}
<   scells     = ${scells:-"2600"}
<   tlives     = ${tlives:-"1800"}
<   nseed      = ${nseed:-"1"}
<   nfracseed  = ${nfracseed:-"0.5"}
<   rcell      = ${rcell:-"0.72"}
<   ca_trigger = ${ca_trigger:-".True."}
<   nspinup    = ${nspinup:-"1"}
<   iseed_ca   = ${ISEED_CA:-"12345"}
< EOF

which I do not see anywhere here.

WalterKolczynski-NOAA replied:
This was just added to coupled branch Tuesday, but it does need to be brought over (maybe with a default of NO instead).

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+  2)
+  cat >> input.nml << EOF
+  gwd_opt      = 2
+  do_ugwp      = .true.
```
This line showed up as a diff from f/c-c but maybe it's okay as is.  @WalterKolczynski-NOAA can you weigh in on that?

WalterKolczynski-NOAA replied:
Should be false. It doesn't make sense to me, but that's what is in the RT.

WalterKolczynski-NOAA replied:
This still needs to be changed back to .false.

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+  FSMCL(2) = ${FSMCL2:-99999}
+  FSMCL(3) = ${FSMCL3:-99999}
+  FSMCL(4) = ${FSMCL4:-99999}
+  LANDICE  = ${landice:-".true."}
```
Default in f/c-c is false

WalterKolczynski-NOAA replied:
This should be set based on the CCPP suite in postdet (depends on the LSM). I've already made this change in the coupled branch. But true is correct for Noah.

WalterKolczynski-NOAA replied:
(I added it just this week)

JessicaMeixner-NOAA commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,603 @@
+
+# Add namelist for stochastic physics options
+echo "" >> input.nml
+if [ $MEMBER -gt 0 ]; then
```
In f/c-c I believe this if statement was copied out and replaced with: 
 if [ $DO_SPPT = .true. -o $DO_SHUM = .true. -o $DO_SKEB = .true. ]; then

jkuang-exploro replied:
Done

WalterKolczynski-NOAA replied:
This still isn't corrected.

jkuang-exploro replied:
OK. I made the changes to another copy of my disk therefore not commit it. Now did it.

WalterKolczynski-NOAA commented:
@JianKuang-UMD Please take a look at the additional changes I made to coupled-crow this week, especially the changes to postdet that are based on the CCPP suite. Some of the defaults in parsing namelist should remain as they are now though. Ask me if you have any questions about a particular setting.

Commit:
update scripts to match coupled_sprint_0806 tag, off from
feature/coupled-crow.

Need to decide if the following default variable update
should go in or not:

lheatstrg  = ${lheatstrg-“.false.“}
lseaspray  = ${lseaspray:-“.false.“}

Commit:
permission setting unification

aerorahul commented:
@JianKuang-UMD 
Is there a reason these templates need to have exec permissions?

aerorahul commented:
@JianKuang-UMD 
There are still trailing whitespaces in the parsing files.

aerorahul commented:
Trailing spaces in the `forecast_*.sh` files too.

jkuang-exploro commented:
@aerorahul Unfortunately I wasn't able to see those trailing whitespaces. And for some reason I am unable to simply push a change for file permissions.

aerorahul committed:
templates .IN are not executables.  Strip trailing whitespaces and expand tabs for mixed indents

aerorahul committed:
fix more trailing spaces in the scripts added

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -0,0 +1,968 @@
+    nfiles=$(ls -1 $DATA/INPUT/* | wc -l)
+    if [ $nfiles -le 0 ]; then
+      echo SUB ${FUNCNAME[0]}: Initial conditions must exist in $DATA/INPUT, ABORT!
+      msg=”"SUB ${FUNCNAME[0]}: Initial conditions must exist in $DATA/INPUT, ABORT!"
```
Looks like a syntax error with two different kinds of quotation marks to start the right side of the assignment.

jkuang-exploro replied:
This is a typo. Now addressing it.

WalterKolczynski-NOAA reviewed (changes requested):
A couple changes that were supposedly made haven't actually been made. Maybe forgot to push some commits? Also one syntax issue.

Commit:
fix typo
update do_ugwp

Commit:
update if logic of MEMBER to DO_SPPT; DO_SHUM, DO_SKEB

aerorahul committed:
fix fcst tasks issue for gdas and gfs. fix ocnres to 050

Commit:
update ENV file for efcs job

Commit:
properly design DIAG_TABLE and DIAG_TABLE_APPEND

Commit:
update stochastic parameters
move OCNRES to config.fv3

aerorahul commented on parm/config/config.fv3:
```diff
@@ -51,6 +51,7 @@ case $case_in in
         export WRITE_GROUP_GFS=1
         export WRTTASK_PER_GROUP_GFS=$npe_node_max
         export WRTIOBUF="1M"
+	export OCNRES=100
```
This is not an appropriate place to set `OCNRES`.
For now, please create a case block in `config.fcst`
Besides, this is only needed if `FRACGRID` is `YES` (I think).

aerorahul replied:
Can you also please pay attention to indentation.

jkuang-exploro replied:
Moved to config.fcst

aerorahul commented on parm/config/config.fv3:
```diff
@@ -69,6 +70,7 @@ case $case_in in
         export WRTTASK_PER_GROUP_GFS=$npe_node_max
         export WRTIOBUF="4M"
         export n_split=6
+	export OCNRES=100
```
Same comment as above and for every change below.

aerorahul commented on ush/parsing_model_configure_DATM.sh:
```diff
@@ -0,0 +1,40 @@
+cat > model_configure <<EOF
+total_member:              ${ENS_NUM:-1}
+print_esmf:                ${print_esmf:-.true.}
+PE_MEMBER01:               $NTASKS_TOT
```
I think this option is gone from the model.

jkuang-exploro replied:
What "option" are we talking about here? Could you please explain to me through email or slack?

JessicaMeixner-NOAA replied:
See my PR #402 has the options that are removed in it. You could look at that for which model_configure options are gone although this is for DATM, so you should probably look at the model_configure template for the DATM in ufs-weahter-model for the most up to date version.

JianKuang-UMD replied:
Do these changes work WITHOUT the hashtag change included in #402?

jkuang-exploro replied:
OK These has been addressed also. I performed a quick test to ensure the model runs OK without total_member and PE_MEMBER01

aerorahul commented on ush/parsing_model_configure_DATM.sh:
```diff
@@ -0,0 +1,40 @@
+
+rm -f model_configure
+cat > model_configure <<EOF
+total_member:              ${ENS_NUM:-1}
```
This option is also gone from the model.

aerorahul commented on ush/parsing_model_configure_FV3.sh:
```diff
@@ -0,0 +1,82 @@
+cat >> model_configure <<EOF
+total_member:            $ENS_NUM
+print_esmf:              ${print_esmf:-.true.}
+PE_MEMBER01:             $NTASKS_TOT
```
Gone.  Please check.

aerorahul commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,619 @@
+# Add namelist for stochastic physics options
+echo "" >> input.nml
+#if [ $MEMBER -gt 0 ]; then
+if [ $do_sppt = .true. -o $do_shum = .true. -o $do_skeb = .true. ]; then
```
I think we agreed on using `DO_SPPT=YES` for script variables logic changes.

WalterKolczynski-NOAA replied:
I really hate having variables with identical names except case that are set to different things.

jkuang-exploro replied:
Understood

aerorahul commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,619 @@
+&nam_stochy
+EOF
+
+  if [ $do_skeb = ".true." ]; then
```
Same comment as above.

aerorahul commented on ush/parsing_namelists_FV3.sh:
```diff
@@ -0,0 +1,619 @@
+EOF
+  fi
+
+  if [ $do_sppt = ".true." ]; then
```
And here.

aerorahul reviewed (changes requested):
Please see comments.

WalterKolczynski-NOAA commented on parm/config/config.fcst:
```diff
@@ -103,6 +103,9 @@ export ICEPETS="80"
 export WAVPETS="160"
 export CHMPETS="1700"
 
+[[ "$CDUMP" = "gfs" ]] && ntasks_fcst=$npe_fcst_gfs || ntasks_fcst=$npe_fcst
```
Please expand this out into a proper if/then/else block. I'm going to be removing all of this sort of short-circuit soon.

aerorahul replied:
Can I ask why?
This is a standard technique, succinct and clear.

jkuang-exploro replied:
What does "short-circuit" mean here? I mean they are equivalent to if/then/else logically.

jkuang-exploro replied:
This is addressed.

WalterKolczynski-NOAA commented on parm/config/config.fcst:
```diff
@@ -337,13 +340,13 @@ elif [[ "$CDUMP" == "gfs" ]] ; then # GFS cycle specific parameters
 
 if [ $DO_AERO = "YES" ]; then # temporary settings for aerosol coupling
-    export DIAG_TABLE="$HOMEgfs/parm/parm_fv3diag/diag_table_aer"
+    export DIAG_TABLE_APPEND="$HOMEgfs/parm/parm_fv3diag/diag_table_aer"
```
diag_table_aer and diag_table_cpl are both full diag tables, not something to be appended.

jkuang-exploro replied:
Understood

WalterKolczynski-NOAA commented on parm/config/config.efcs:
```diff
@@ -18,6 +18,7 @@ if [ $QUILTING = ".true." ]; then
     export npe_efcs=$npe_fv3
 fi
 
+export ntasks_efcs=$npe_efcs
```
Why are we introducing a new variable that is exactly the same?

jkuang-exploro replied:
I update it similar with config.fcst. In order to pass ntasks to env file.

jkuang-exploro replied:
addressed

Commit:
address changes requested by group members
move OCNRES to config.fcst
use DO_* switches to control parsing_namelists_FV3.sh
eliminate duplicated variable ntasks_efcs

WalterKolczynski-NOAA commented:
> @WalterKolczynski-NOAA
> Please explain why one-liners need to be replaced with explicit if-then-else blocks?
> 
> Does it improve readability?
> Is there a flaw in the execution?
> 
> By extension, will the use of ternary operators be disallowed?

Improved readability. We can discuss ternary operators as a group, though I don't have a big issue with them since there are at least clearly assignments and they are designed to do that. But I also know I had trouble with them once upon a time, so I wouldn't say no to prohibiting them to keep code more readable. Bash doesn't have them anyway.

Commit:
remove obsolete parameters PE_MEMBER1 and total_member
